src/__tests__/algorand.test.ts

```diff
@@ -92,6 +92,27 @@ describe('AlgorandApp', () => {
     })
   })
 
+  describe('getAddressAndPubKey', () => {
+    it('should return public key and address', async () => {
+      const mockResponse = Buffer.from([
+        // Mock public key (32 bytes)
+        ...Array(32).fill(1),
+        // Mock address (58 bytes)
+        ...Array(58).fill(2),
+        // Return code (2 bytes)
+        0x90,
+        0x00,
+      ])
+      mockSend.mockResolvedValue(mockResponse)
+
+      const result = await app.getAddressAndPubKey()
+      expect(result).toBeDefined()
+      expect(result.publicKey).toBeDefined()
+      expect(result.address).toBeDefined()
```

Comment:
We could improve this, and compare to the fix value we set on the mock transport. We could test we are parsing the response correctly (pk and address)